pkg/file/delete.go

```diff
@@ -58,20 +58,33 @@ func newRenamerRemoverImpl() renamerRemoverImpl {
 
 // Deleter is used to safely delete files and directories from the filesystem.
 // During a transaction, files and directories are marked for deletion using
-// the Files and Dirs methods. This will rename the files/directories to be
-// deleted. If the transaction is rolled back, then the files/directories can
-// be restored to their original state with the Abort method. If the
-// transaction is committed, the marked files are then deleted from the
-// filesystem using the Complete method.
+// the Files and Dirs methods. If TrashPath is set, files are moved to trash
+// immediately. Otherwise, they are renamed with a .delete suffix. If the
+// transaction is rolled back, then the files/directories can be restored to
+// their original state with the Rollback method. If the transaction is
+// committed, the marked files are then deleted from the filesystem using the
+// Commit method.
 type Deleter struct {
 	RenamerRemover RenamerRemover
 	files          []string
 	dirs           []string
+	TrashPath      string            // if set, files will be moved to this directory instead of being permanently deleted
+	trashedPaths   map[string]string // map of original path -> trash path (only used when TrashPath is set)
 }
 
 func NewDeleter() *Deleter {
 	return &Deleter{
 		RenamerRemover: newRenamerRemoverImpl(),
+		TrashPath:      "",
+		trashedPaths:   make(map[string]string),
+	}
+}
+
+func NewDeleterWithTrash(trashPath string) *Deleter {
+	return &Deleter{
+		RenamerRemover: newRenamerRemoverImpl(),
+		TrashPath:      trashPath,
+		trashedPaths:   make(map[string]string),
 	}
 }
```

Comment:
`NewDeleter` is now unused. Perhaps you should just make NewDeleter accept the trash path.